src/claude_client.rs

```diff
@@ -0,0 +1,333 @@
+    let context_data = {
+      let context = self.get_or_create_conversation(conversation_key);
+      context.clone()
+    };
+
+    let response = self.call_claude_cli(&context_data).await?;
+
+    // Finally, add the assistant response
+    {
+      let context = self.get_or_create_conversation(conversation_key);
+      context.add_message("assistant", &response.content);
+    }
+
+    // Save the updated conversation to persistence
+    self.save_conversation_to_persistence(conversation_key);
+
+    Ok(response.content)
+  }
+
+  pub async fn send_message_with_discord_history(
+    &mut self,
+    conversation_key: &ConversationId,
+    discord_messages: &[(String, String)], // (author_name, content) pairs
+    current_user_message: &str,
+  ) -> Result<String> {
+    // If no Discord history, use the simple send_message method
+    if discord_messages.is_empty() {
+      return self
+        .send_message(conversation_key, current_user_message)
+        .await;
+    }
+
+    // Get or create conversation context and initialize with Discord history if new
+    {
+      let context = self.get_or_create_conversation(conversation_key);
+
+      // If this is a new conversation, initialize it with Discord history
+      if context.messages.is_empty() {
+        for (author_name, content) in discord_messages {
+          if author_name.starts_with("bot:") {
+            context.add_message("assistant", content);
+          } else {
+            context.add_message("user", &format!("User {} says: {}", author_name, content));
+          }
+        }
+      }
+
+      // Add the current message
+      context.add_message("user", current_user_message);
+    }
+
+    // Clone the context data for the Claude call
+    let context_data = {
+      let context = self.get_or_create_conversation(conversation_key);
+      context.clone()
+    };
+
+    let response = self.call_claude_cli(&context_data).await?;
+
+    // Add the assistant response
+    {
+      let context = self.get_or_create_conversation(conversation_key);
+      context.add_message("assistant", &response.content);
+    }
+
+    // Save the updated conversation to persistence
+    self.save_conversation_to_persistence(conversation_key);
+
+    Ok(response.content)
+  }
+
+  fn save_conversation_to_persistence(&self, conversation_key: &ConversationId) {
+    if let Some(context) = self.conversations.get(conversation_key) {
+      if let Err(e) = self
+        .persistence
+        .save_conversation(conversation_key, context)
+      {
+        error!(
+          "Failed to save conversation {} to persistence: {}",
+          conversation_key, e
+        );
+      }
+    }
+  }
+
+  async fn call_claude_cli(&self, context: &ConversationContext) -> Result<ClaudeResponse> {
+    let claude_path = self
+      .claude_path
+      .as_ref()
+      .ok_or_else(|| anyhow!("Claude CLI not available - chat mode is disabled"))?;
```

Comment:
Perhaps we can cache this value with a `Lazy` rather than store an `Option` on `self`. This way, we only look this path up when it is necessary rather than on application start up (should someone want to default the bot to having chat-mode disabled)